File: OddOccurrencesInArray/OddOccurrencesInArray.java
import java.util.*;

class Solution {
    public int solution(int[] A) {
        Set<Integer> found = new HashSet<Integer>();
        for (Integer a : A) {
            if (found.contains(a)) {
                found.remove(a);
            } else {
                found.add(a);
            }
        }
        return found.stream().findFirst().orElseGet(null);
    }
}

File: OddOccurrencesInArray/OddOccurrencesInArray.py
def solution(A):
    found = set()
    for a in A:
        if a in found:
            found.remove(a)
        else:
            found.add(a)
    return found.pop()

# https://app.codility.com/demo/results/trainingXY4E29-3CC/
